lib/Crypto/DecryptAll.php

```diff
@@ -90,55 +90,77 @@ public function __construct(
 	 * @throws LoginException
 	 */
 	public function prepare(InputInterface $input, OutputInterface $output, $user) {
+		// We need to setup the module with everything it needs in order to decrypt user data
+		// This could be using master key or user key encryption
+		// and could also be for a single user or multiple users
 
-		$question = new Question('Please enter the recovery key password: ');
-
+		// First lets work out the mode
 		if ($this->util->isMasterKeyEnabled()) {
-			$output->writeln('Use master key to decrypt all files');
+			$output->writeln('Using master key for decryption');
 			$user = $this->keyManager->getMasterKeyId();
 			$password = $this->keyManager->getMasterKeyPassword();
 		} else {
-			$recoveryKeyId = $this->keyManager->getRecoveryKeyId();
-			if (!empty($user)) {
-				$output->writeln('You can only decrypt the users files if you know');
-				$output->writeln('the users password or if he activated the recovery key.');
-				$output->writeln('');
-				$questionUseLoginPassword = new ConfirmationQuestion(
-					'Do you want to use the user: ' . $user . ' login password to decrypt all files? (y/n) ',
-					false
-				);
-				$useLoginPassword = $this->questionHelper->ask($input, $output, $questionUseLoginPassword);
-				if ($useLoginPassword) {
-					$question = new Question('Please enter the user\'s login password: ');
-				} else if ($this->util->isRecoveryEnabledForUser($user) === false) {
-					$output->writeln('No recovery key available for user ' . $user);
-					return false;
+			$throwLoginException = false;
+			$output->writeln("Configuring encryption module for decryption with user based keys");
+			// We must be using user key - now finish the prep
+			// Must either be using recovery keys, or just decrypting one user and know their password
+
+			// Check a method has been passed
+			if(!$input->hasOption('method') && !in_array($input->getOption('method'),['recovery', 'password'])) {
+				$output->writeln('A method must be supplied when decrypting from user-key state');
+				return false;
+			}
+			if (empty($user)) {
+				// all users, so only recovery is possible
+				if($input->getOption('method')==='recovery' && !empty(getenv('OC_RECOVERY_PASSWORD'))) {
```

Comment:
please avoid using `empty` because in PHP `empty("0")` is true...

Reply:
also please add more spacing

Reply:
and you missed the backslash before `getenv` here

Reply:
Addressed the spacing, backslash before `getenv` and removed usage of `empty`.